homeassistant/components/romy/vacuum.py

```diff
@@ -55,32 +53,26 @@ async def async_setup_entry(
     """Set up ROMY vacuum cleaner."""
 
     coordinator: RomyVacuumCoordinator = hass.data[DOMAIN][config_entry.entry_id]
-    async_add_entities([RomyVacuumEntity(coordinator, coordinator.romy)], True)
+    async_add_entities([RomyVacuumEntity(coordinator)], True)
 
 
-class RomyVacuumEntity(CoordinatorEntity[RomyVacuumCoordinator], StateVacuumEntity):
+class RomyVacuumEntity(
+    RomyEntity, CoordinatorEntity[RomyVacuumCoordinator], StateVacuumEntity
+):
     """Representation of a ROMY vacuum cleaner robot."""
 
-    _attr_has_entity_name = True
-    _attr_name = None
     _attr_supported_features = SUPPORT_ROMY_ROBOT
     _attr_fan_speed_list = FAN_SPEEDS
+    _attr_name = None
 
     def __init__(
         self,
         coordinator: RomyVacuumCoordinator,
-        romy: RomyRobot,
     ) -> None:
         """Initialize the ROMY Robot."""
-        super().__init__(coordinator)
-        self.romy = romy
+        RomyEntity.__init__(self, coordinator.romy)
+        CoordinatorEntity.__init__(self, coordinator)
```

Comment:
```suggestion
        super().__init__(coordinator)
```

Reply:
thats a nice cleanup didn't like that already somehow